pulp_rpm/app/modulemd.py

```diff
@@ -68,7 +68,14 @@ def split_modulemd_file(file):
         cr.decompress_file(file.path, decompressed_path, cr.AUTO_DETECT_COMPRESSION)
         with open(decompressed_path) as modulemd_file:
             module = ""
+            first = True
             for line in modulemd_file:
+                # some modulemd documents, unfortunately, don't start with a document separator.
+                # that is valid yaml although most linters will correct it.
+                if first:
+                    if not line.startswith("---"):
+                        module = "---\n"
+                    first = False
```

Comment:
I don't love this approach so if anyone has suggestions on a better one, I'm open

Reply:
The other options I could think of revolve around checking at yield-time that module.startswith("---") and jamming it in at the beginning if not, which is objectively worse.

You could start the loop with something like `is_separator == line.startswith("---")`, and then use that at `if first and not is_separator:` and at line-79. It  *may be* a little cleaner?

Reply:
it's readable so i am ok with that, maybe rename to a more explicit name `first_line`